lib/diagnostic.rb

```diff
@@ -43,6 +56,8 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.open(filename) do |file|
+      file.split('\n').join()
+    end
```

Comment:
You're thinking along the right lines, but that's not how `File.open` works. When you give a block to `File.open`, the thing that the block will be passed is a **line of text** from the file, not the entire contents.

`File.open(filename).map { | line | line.chomp }.join(' ').reduce(&:+)` is one way to solve this.